haicrypt/hcrypt_ctx_rx.c

```diff
@@ -164,7 +164,7 @@ int hcryptCtx_Rx_ParseKM(hcrypt_Session *crypto, unsigned char *km_msg, size_t m
 	}
 
 	/* Unwrap SEK(s) and set in context */
-	if (0 > hcrypt_UnwrapKey(&ctx->aes_kek, seks,
+	if (0 < hcrypt_UnwrapKey(&ctx->aes_kek, seks,
```

Comment:
The function returns the length of the unwrapped key when successful. This change also affect the other (openssl) implementation. Making the SA (Security Association) then decryption impossible). The original code is also wrong since 0 is returned for a decryption with the wrong key (wrong passphrase). The condition should be (0 >= hcrypt_Unwrap.... With this error I doubt a successfull encrypted session have ever been established with this implementation.

Reply:
> The function returns the length of the unwrapped key when successful. This change also affect the other (openssl) implementation.

You're right. Actually, I tended to send a bug fix for this error checking condition.
As far as I understand,  `>` and `>=` mean that it returns `(-2)` when `hcrypt_UnwrapKey` is done successfully. If we need to return `(-2)` in case of failure, I thought it should be `<`.

Did I miss something else? To raise this issue again, I'll create another issue and remove this commit from my PR.

Reply:
I opened #151 to discuss more. Thanks!